src/journal/journal-send.c

```diff
@@ -73,33 +73,52 @@ _public_ int sd_journal_print(int priority, const char *format, ...) {
 }
 
 _public_ int sd_journal_printv(int priority, const char *format, va_list ap) {
-
-        /* FIXME: Instead of limiting things to LINE_MAX we could do a
-           C99 variable-length array on the stack here in a loop. */
-
-        char buffer[8 + LINE_MAX], p[STRLEN("PRIORITY=") + DECIMAL_STR_MAX(int) + 1];
-        struct iovec iov[2];
+        char p[STRLEN("PRIORITY=") + DECIMAL_STR_MAX(int) + 1];
+        int len, i, nlines;
+        _cleanup_free_ char *tbuffer;
+        char *s, *d;
 
         assert_return(priority >= 0, -EINVAL);
         assert_return(priority <= 7, -EINVAL);
         assert_return(format, -EINVAL);
 
         xsprintf(p, "PRIORITY=%i", priority & LOG_PRIMASK);
 
-        memcpy(buffer, "MESSAGE=", 8);
-        vsnprintf(buffer+8, sizeof(buffer) - 8, format, ap);
-
-        /* Strip trailing whitespace, keep prefix whitespace. */
-        (void) strstrip(buffer);
-
-        /* Suppress empty lines */
-        if (isempty(buffer+8))
+        len=vasprintf(&tbuffer, format, ap);
+        if (len <= 0)
                 return 0;
-
-        iov[0] = IOVEC_MAKE_STRING(buffer);
-        iov[1] = IOVEC_MAKE_STRING(p);
-
-        return sd_journal_sendv(iov, 2);
+        nlines = len/LINE_MAX;
+        for (;;) {
+                struct iovec iov[nlines + 2];
+                char buffer[8 + len];
+                memzero(buffer, sizeof(buffer));
+                memcpy(buffer, "MESSAGE=", 8);
```

Comment:
There's no reason to use memzero and memcpy when plain initialization is good enough:
```C
char buffer[8 + len] = "MESSAGE=";
```

Reply:
Variable length array can't be initialised by direct assignment.